dmscripts/supplier_data.py

```diff
@@ -0,0 +1,103 @@
+
+    def get_supplier_users(self):
+        """Return a dict, supplier id: application status."""
+        users = self.client.export_users(self.target_framework_slug).get('users', [])
+        sort_by_func = lambda i: i['supplier_id']
+        sorted_users = sorted(users, key=sort_by_func)
+        return {k: list(g) for k, g in groupby(sorted_users, key=sort_by_func)}
+
+    def get_supplier_service_data(self, supplier_id):
+        """Given a supplier ID return a list of dictionaries for services related to framework."""
+        return self.client.find_draft_services_iter(supplier_id, framework=self.target_framework_slug)
+
+    def populate_data(self):
+        """Populate a dict with supplier data from the api."""
+        self.data = self.get_supplier_frameworks()
+        users = self.get_supplier_users()
+        for supplier_framework in self.data:
+            supplier_id = supplier_framework['supplierId']
+            supplier_framework['users'] = users.get(supplier_id, [])
+            supplier_framework['services'] = list(self.get_supplier_service_data(supplier_id))
+
+
+class SuccessfulSupplierContextForNotify(SupplierFrameworkData):
+    """Get the personalisation/ context for 'Your application result - if successful email'"""
+
+    STATUS_MAP = {'submitted': 'Successful', 'not-submitted': 'No application', 'failed': 'Unsuccessful'}
+
+    def __init__(self, client, target_framework_slug):
+        """Get the target framework to operate on and list the lots.
+
+        :param client: Instantiated api client
+        :param target_framework_slug: Framework to fetch data for
+        """
+        self.date_today = date.today().strftime(DISPLAY_DATE_FORMAT)
+        super(SuccessfulSupplierContextForNotify, self).__init__(client, target_framework_slug)
+
+        self.framework = client.get_framework(self.target_framework_slug)['frameworks']
+        self.framework_lots = [i['name'] for i in self.framework['lots']]
+
+    def get_users_peronalisations(self):
+        """Return {email_address: {personalisations}} for users eligible for the
+        'Your application result - if successful' email
+        """
+        output = {}
+        for supplier_framework in filter(lambda i: i['onFramework'], self.data):
+            for user in supplier_framework['users']:
+                output.update(self.get_user_personalisation(user, supplier_framework))
+        return output
+
+    def get_lot_dict(self, supplier_framework):
+        """Return a dict of lot status for each lot on the framework.
+        """
+        any_lots = False
+
+        lot_dict = OrderedDict((lot_name, 'No application') for lot_name in self.framework_lots)
+        for service in supplier_framework['services']:
+            status = service['status']
+            if status == 'submitted':
+                any_lots = True
+            lot_dict[service['lotName']] = self.STATUS_MAP[status]
```

Comment:
This works for the "one-service-per-lot" lots, but for the `user-research-studios` lot it's possible to have multiple services for the lot, and so maybe some submitted and some not-submitted services in the lot.  If there's a single `submitted` user-research-studio then the application will be `Successful` overall.

(And in general - e.g. all G-Cloud lots - we should expect that there could be multiple draft services per lot, some submitted and some not.)